File: 274-h-index/h-index.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        sort(citations.begin(), citations.end());
        int count = 0;
        int n =  citations.size();
        for(int i = n-1; i>= 0 ; i--){
            if(count>=citations[i]) return count;
            count++;
        }
        return count;
    }
};

File: 274-h-index/h-index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        minn = 0
        maxx = len(citations)
        ans = 0
        while(minn<=maxx):
            avgg = (minn + maxx)//2
            count = 0
            for i in citations:
                if i >= avgg: count+=1
            if count >= avgg : 
                ans = avgg
                minn = avgg+1
            else : 
                maxx = avgg-1
        return ans